### PR TITLE
feat(core): refactor `Dropdown` to not require `HostedDropdown`

### PR DESCRIPTION
**Pull request by waterplea**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?

Closes #

### CODE REVIEW

File: projects/core/components/hosted-dropdown/dropdown-open-monitor.directive.ts [hunk 1]
```diff
@@ -22,7 +22,10 @@ export class TuiDropdownOpenMonitorDirective {
     ) {
         open?.tuiDropdownOpenChange
             .pipe(
-                filter(value => value && open.dropdown === dropdown && !hosted.focused),
+                filter(
+                    value =>
+                        value && open.dropdown === (dropdown as any) && !hosted.focused,
```
Comment (waterplea) [inline projects/core/components/hosted-dropdown/dropdown-open-monitor.directive.ts:27]: This whole file will be removed soon

**Review by splincode — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/core/directives/dropdown/dropdown-position.directive.ts [hunk 2]
```diff
@@ -10,27 +9,22 @@ import {TUI_VIEWPORT} from '@taiga-ui/core/tokens';
...
+@Directive({standalone: true, selector: '[tuiDropdownPosition]'})
 export class TuiDropdownPositionDirective extends TuiPositionAccessor {
+    private readonly options = inject(TUI_DROPDOWN_OPTIONS);
+    private readonly viewport = inject(TUI_VIEWPORT);
+
     private previous?: TuiVerticalDirection;
 
     readonly type = 'dropdown';
+    readonly accessor = tuiFallbackRectAccessor('dropdown')(
+        inject(TuiRectAccessor) as any as readonly TuiRectAccessor[],
```
Comment (nsbarsukov) [inline projects/core/directives/dropdown/dropdown-position.directive.ts:23]: 😢

File: projects/demo/src/modules/app/pages.ts [hunk 3]
```diff
@@ -284,6 +284,12 @@ export const pages: TuiDocPages = [
                 keywords: `dropdown, контекст, выпадашка, дропдаун, Context`,
                 route: `/directives/dropdown`,
             },
+            {
+                section: `Tools`,
+                title: `DropdownOpen`,
+                keywords: `dropdown, контекст, выпадашка, дропдаун, меню, menu`,
```
Comment (nsbarsukov) [inline projects/demo/src/modules/app/pages.ts:290]: Should we add keyword "hosted" (for users who are already used to the previous name)?

**Review by nsbarsukov — APPROVED** [review]

File: projects/core/directives/dropdown/dropdown-position.directive.ts (hunk 2 above)
Comment (waterplea) [reply projects/core/directives/dropdown/dropdown-position.directive.ts:23]: Unfortunately that's the "official" way to handle multi token providers and `inject`

**Pull request merged**